### ISSUE TITLE
Add tests for streaming

### ISSUE THREAD

**Issue by dlongley** [digitalbazaar/edv-client#11]

**Comment by aljones15** [comment]
These tests have been started here: https://github.com/digitalbazaar/edv-client/tree/add-stream-tests

just in case anyone else anyone ends up working on this. at the moment the one insert a stream test is passing in node and karma.

**Comment by mattcollier** [comment]
We need this in support of converting to use http-client.  https://github.com/digitalbazaar/edv-client/issues/69

**Comment by aljones15** [comment]
So references and helpers:

the `minimal-cipher` test suite contains [stream tests](https://github.com/digitalbazaar/minimal-cipher/blob/ee3bdcf3d1942bdb9b24382c1c8d3a5b0def4708/test/unit/index.js#L76-L158)

Streams are a newer tech in which data is broken up into [binary chunks](https://developer.mozilla.org/en-US/docs/Web/API/Streams_API)

in `minimal-cipher` and `edv` there is a `chunkIndex`. In the edv client the `chunkIndex` is determined by the [`chunkSize`](https://github.com/digitalbazaar/edv-client/blob/943b964d564d52b4561136685d7615b09e7aacf0/EdvClient.js#L14) and the size of the file that is being streamed. Our `chunkSize` defaults to 1 MB so a 2.5 MB file would start with a `chunkIndex` of 0 ending with a `chunkIndex` of 2. When you stream the chunk back to the browser this is what happens:

https://github.com/digitalbazaar/edv-client/blob/943b964d564d52b4561136685d7615b09e7aacf0/EdvClient.js#L458-L490


So basically the chunk `/get` mock endpoint will need to accept a `doc` and  `chunkIndex` and then return the chunk that matches that `doc.id` and `chunkIndex`. That chunk is probably a `unit8Array` that is part of an encrypted stream. You can see how the encrypted streams work with `minimal-cipher`.
